File: PHP/30DaysOfCode/Sorting.php
<?php

$handle = fopen ("php://stdin", "r");
fscanf($handle, "%d",$n);
$array = fgets($handle);
$array = explode(" ",$array);
$array = array_map('intval', $array);

bubbleSort($array);

function bubbleSort($array){
    $numSwaps = 0;
    for($i=0; $i<count($array); $i++){
        for($j=0; $j<count($array)-1; $j++){
            if($array[$j] > $array[$j+1]){
                $temp = $array[$j];
                $array[$j] = $array[$j+1];
                $array[$j+1] = $temp;
                $numSwaps++;
            }
        }
        if($numSwaps == 0){
            break;
        }
    }
    
    echo "Array is sorted in ".$numSwaps." swaps.\n";
    echo "First Element: ".$array[0]."\n";
    echo "Last Element: ".$array[count($array)-1]."\n";
    
}



?>

File: JAVA/src/daysOfCode30/day20/Sorting.java
package daysOfCode30.day20;

import java.util.*;

public class Sorting {
	private static int[] array;
	private static int n;
	private static int numSwaps = 0;
	
	public static void bubleSort(){
        for(int i=0; i<n; i++){
        	for(int j=0; j<n-1; j++){
        		if(array[j]>array[j+1]){
	        		int temp = array[j];
	        		array[j] = array[j+1];
	        		array[j+1] = temp;
	        		numSwaps++;
        		}
        	}
        	if(numSwaps == 0){
        		break;
        	}
        }
	}
	
	
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        n = in.nextInt();
        array = new int[n];
        for(int i=0; i < n; i++){
            array[i] = in.nextInt();
        }

        bubleSort();
        
        System.out.println("Array is sorted in " + numSwaps + " swaps.");
        System.out.println("First Element: " + array[0]);
        System.out.println("Last Element: " + array[n-1]);
        
    }
	
	
}
